clang-tools-extra/clang-tidy/tool/clang-tidy-diff.py

```diff
@@ -243,11 +264,17 @@ def main():
 
     args = parser.parse_args(argv)
 
+    # Read compile_commands.json database to extract a compiling file list
+    current_dir = Path.cwd()
+    compile_commands_json = (current_dir / args.build_path) if args.build_path else current_dir
+    compile_commands_json = (compile_commands_json / "compile_commands.json")
+    compiling_files = get_compiling_file_list(compile_commands_json)
+
```

Comment:
If the option is not enabled, this is doing unnecessary work, though this won't show up as problematic in measurements.

You could refactor this part into a function and do:
```python
compiling_files = get_compiling_files(args) if args.skip_non_compiling else None

...

# Skip any files not in the compiling list
if compiling_files is not None and (current_dir / filename) not in compiling_files:
    continue
```